### PR TITLE
fix(availability-sync): detect deleted seasons when media server retains empty season metadata

### PR DESCRIPTION
**Pull request by fallenbagel**
## Description

The availability sync was not detecting deleted seasons on Jellyfin
(and potentially Plex), causing them to stay marked as `AVAILABLE` even
after all episode files were removed from the media server. Users were
unable to re-request these seasons.

The issue was that Jellyfin's `/Shows/{id}/Seasons` endpoint sometimes returned
season metadata entries even when every episode in that season has been
deleted. The season existence check only looked for a matching season
metadata entry, so it always returned true for every season the media
server knew about, regardless of whether any files actually existed.
This overrode the correctly computed "not available" state from the
filtered seasons map, preventing the season updater from ever firing.

Debug logging on a similar scenario confirmed this for a Show instance (TMDB 1408):
Jellyfin returned all 9 season entries, Sonarr correctly reported only
season 6, but the Jellyfin result overwrote everything in the final
seasons map.

The fix adds an episode-level check in both `seasonExistsInJellyfin`
and `seasonExistsInPlex`. After finding a season metadata entry, the
sync now fetches the season's episodes and confirms at least one
non-virtual episode exists before marking the season as available.
On API error, it defaults to assuming the season exists to avoid
false removal.

- Supersedes ##2318

## How Has This Been Tested?
- This was tested by a user on discord and confirmed as working (Jellyfin)
- Not tested on Plex instance but unit tests passes

## Checklist:
- [X] I have read and followed the contribution [guidelines](https://github.com/seerr-team/seerr/blob/develop/CONTRIBUTING.md).
- [ ] Disclosed any use of AI (see our [policy](https://github.com/seerr-team/seerr/blob/develop/CONTRIBUTING.md#ai-assistance-notice))
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] Successful build `pnpm build`
- [ ] Translation keys `pnpm i18n:extract`
- [ ] Database migration (if required)

## Summary by CodeRabbit

* **Bug Fixes**
  * More accurate season existence checks for Plex and Jellyfin by verifying actual episode data per season, with per-sync caching and safe fallbacks that assume existence if episode retrieval fails to avoid accidental removals; handles cases where Sonarr linkage is absent.

* **Tests**
  * Added comprehensive availability sync tests covering Jellyfin, Plex, and Sonarr scenarios and edge cases.

### CODE REVIEW

File: server/lib/availabilitySync.test.ts [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+function fakePlexEpisodes(count: number): PlexMetadata[] {
+  return Array.from({ length: count }, (_, i) => ({
+    ratingKey: `ep-${i}`,
+    guid: `plex://episode/ep-${i}`,
+    type: 'movie' as const,
+    title: `Episode ${i + 1}`,
+    Guid: [],
+    index: i + 1,
+    leafCount: 0,
+    viewedLeafCount: 0,
```
[the remark this answers is not in the document]
Comment (fallenbagel) [reply server/lib/availabilitySync.test.ts:236]: `PlexMetadata`'s type is `'movie' | 'show' | 'season'` and it doesn't include 'episode'. So 'movie' is the closest available value in the union. `PlexMetadata` typing doesn't cover episodes. Not worth fixing in this PR though especially since it's a test helper and the code only checks length.

**Review by gauthier-th — APPROVED** [review]
LGTM

**Review by 0xSysR3ll — APPROVED** [review]
